File: backend/app/schemas/usuario.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, EmailStr


class UsuarioCreate(BaseModel):
    nome: str
    perfil: Literal["gestor", "recepcionista", "medico", "paciente"]
    login: str
    email: EmailStr
    password: str
    status: Literal["ativo", "inativo"] = "ativo"
    modulos_permitidos: list[str] | None = None
    observacao: str | None = None


class UsuarioUpdate(BaseModel):
    nome: str | None = None
    perfil: Literal["gestor", "recepcionista", "medico", "paciente"] | None = None
    email: EmailStr | None = None
    status: Literal["ativo", "inativo"] | None = None
    modulos_permitidos: list[str] | None = None
    observacao: str | None = None
    password: str | None = None


class UsuarioResponse(BaseModel):
    id: int
    nome: str
    perfil: str
    login: str
    email: str
    status: str
    ultimo_acesso: datetime | None
    modulos_permitidos: list | None
    observacao: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/usuario.ts
import type { UserPerfil, UserStatus } from './auth';

export interface Usuario {
  id: number;
  nome: string;
  perfil: UserPerfil;
  login: string;
  email: string;
  status: UserStatus;
  ultimo_acesso: string | null;
  modulos_permitidos: string[] | null;
  observacao: string | null;
  created_at: string;
}

export interface UsuarioCreate {
  nome: string;
  perfil: UserPerfil;
  login: string;
  email: string;
  password: string;
  status?: UserStatus;
  modulos_permitidos?: string[];
  observacao?: string;
}

export interface UsuarioUpdate {
  nome?: string;
  perfil?: UserPerfil;
  email?: string;
  status?: UserStatus;
  modulos_permitidos?: string[];
  observacao?: string;
  password?: string;
}

export interface LogAuditoria {
  id: number;
  data_hora: string;
  usuario_id: number | null;
  acao: string;
  modulo: string;
  ip: string | null;
  resultado: 'sucesso' | 'falha';
  detalhes: string | null;
}
